### PR TITLE
[PRO-1913] add scope checking to everything

### PR DESCRIPTION
**Pull request by Danten**

### CODE REVIEW

File: common/src/main/scala/org/genivi/sota/http/Scopes.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  lazy val updates_str = apiDomain + "updates"
+  def updates(ns: AuthedNamespaceScope) = new ScopeDirectives(ns, updates_str)
+}
+
+class ScopeDirectives(ns: AuthedNamespaceScope, theScope: String) {
+  def check: Directive0 = ns.oauthScopeReadonly(theScope)
+  def delete: Directive0 = Directives.delete & check
+  def get: Directive0 = Directives.get & check
+  def put: Directive0 = Directives.put & check
+  def post: Directive0 = Directives.post & check
```
Comment (jerrytrieu) [inline common/src/main/scala/org/genivi/sota/http/Scopes.scala]: Can check for write?

File: core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala [hunk 2]
```diff
@@ -260,12 +260,6 @@ class DeviceUpdatesResource(db: Database,
           authedNs.oauthScopeReadonly(s"ota-core.${device.show}.read") {
             downloadPackage(device, updateId)
           }
-        } ~
-        authDeviceNamespace(device) { ns =>
-          path("queued") { pendingPackages(device) } ~
-          path("blocked") { getBlockedInstall(device) } ~
-          path("results") { results(device) } ~
-          (extractRefinedUuid & path("results")) { updateId => resultsForUpdate(device, updateId) }
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala]: Separate commit for this?

Comment (Danten) [inline core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala]: Sure

File: device-registry/src/main/scala/org/genivi/sota/device_registry/DevicesResource.scala [hunk 3]
```diff
@@ -91,34 +90,30 @@ class DevicesResource(namespaceExtractor: Directive1[AuthedNamespaceScope],
...
-          (delete & pathEnd) {
-            deleteDevice(ns, uuid)
-          }
-        }
-      } ~
+      (scope.post & entity(as[DeviceT]) & pathEndOrSingleSlash) { device => createDevice(ns, device) } ~
+      (scope.get & pathEnd) { searchDevice(ns) } ~
       deviceNamespaceAuthorizer { uuid =>
-        (post & path("ping")) {
+        (scope.put & entity(as[DeviceT]) & pathEnd) { device =>
```
Comment (Danten) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/DevicesResource.scala]: Sadly yes, a lot of our extractUuid stuff will extract it twice.

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala [hunk 4]
```diff
@@ -86,44 +86,43 @@ class GroupsResource(namespaceExtractor: Directive1[AuthedNamespaceScope], devic
...
-        (post & path("from_attributes") & extractCreateGroupRequest) { request =>
-          createGroupFromDevices(request, ns)
-        } ~
-        parameter('groupName.as[Name]) { groupName =>
-          (post & path("group_info") & entity(as[Json])) { body =>
-            createGroupInfo(Uuid.generate(), groupName, ns, body)
-          } ~
-          (post & pathEnd) { createGroupInfo(Uuid.generate(), groupName, ns, Json.Null) }
+    (pathPrefix("device_groups") & namespaceExtractor) { ns =>
+      val scope= Scopes.devices(ns)
```
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupsResource.scala]: space here between scope and '='?

**Comment by CalumMcCall** [comment]
11/11

**Pull request merged**